src/main/java/com/uu_demo/models/entity/User.java

```diff
@@ -37,13 +38,14 @@
     private String password;
 
     @NotNull
-    @ManyToMany(fetch = FetchType.LAZY, targetEntity = Role.class)
-    @JoinTable(name = "user_roles", joinColumns = @JoinColumn(name = "user_id"),
-            inverseJoinColumns = @JoinColumn(name = "role_id"))
-    private Set<Role> roles;
+    @ManyToOne(fetch = FetchType.LAZY, targetEntity = Role.class)
```

Comment:
Consider using `FetchType.EAGER` I think you'll almost always need the user's role.